internal/pkg/skuba/deployments/deployments.go

```diff
@@ -44,6 +44,28 @@ type Target struct {
 }
 
 func (t *Target) Apply(data interface{}, states ...string) error {
+	// Mechanism for removing states without changing the order
+	remove := func(s []string, i int) []string {
+		s[len(s)-1], s[i] = s[i], s[len(s)-1]
+		return s[:len(s)-1]
+	}
+
+	// Find empty states
+	var emptyStates []int
+	emptyStateFound := false
+	for i, state := range states {
+		if state == "" {
+			emptyStates = append(emptyStates, i)
+			emptyStateFound = true
+		}
+	}
+
+	// Remove empty states so they don't get applied
+	if emptyStateFound {
+		for _, emptyState := range emptyStates {
+			states = remove(states, emptyState)
+		}
+	}
```

Comment:
You could just filter the empty states like this:

```suggestion
    filteredStates := []string{}
    for _, s := range states {
        if s == "" {
            continue
        }   
        filteredStates = append(filteredStates, s)
    }
    return t.Actionable.Apply(data, newStates...)
```

or maybe it would make more sense to do this filtering inside of the `t.Actionale.Apply` function...

Reply:
I love this. Less is more :D